tooling-docker/src/main/java/org/jboss/eap/qe/ts/common/docker/Docker.java

```diff
@@ -77,7 +78,17 @@ protected void start() throws Exception {
             }
         });
 
-        containerReadyCondition.waitUntilReady(containerReadyTimeout);
+        long startTime = System.currentTimeMillis();
+        while (!(new CompletableFuture().supplyAsync(() -> containerReadyCondition.isReady())).get()) {
```

Comment:
Just a curious question: Why is this async when there is a waiting for result?

Reply:
I did some tries to specify timeout on isReady() method so in case it's badly implemented, it does not hang forever and `containerReadyTimeout` will be applied however I did not finish it. I've changed the code so it's applied and added test for it.

Now when containerReadyCondition.isReady() will hang then after `containerReadyTimeout` new exception will be thrown.

Reply:
I'll mark this as resolved.